### PR TITLE
osrm-backend: support for mac

### LINKED ISSUE #224535: Support building `osrm-backend` on macOS
### Steps To Reproduce

1. `NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1 nix build nixpkgs#osrm-backend --impure`

### Build log

```
error: builder for '/nix/store/phyfv3vv06g2qjm46f8ipvql840bcjva-luabind-0.9.1.drv' failed with exit code 1;
       last 10 log lines:
       > Jamroot:16: Unescaped special character in argument [luabind
       > Jamroot:16: Unescaped special character in argument DEBUG]
       > Jamroot:93: Unescaped special character in argument prefix:
       > Jamroot:94: Unescaped special character in argument includes:
       > Jamroot:95: Unescaped special character in argument lib:
       > Jamroot:96: Unescaped special character in argument candidates:
       > Jamroot:97: Unescaped special character in argument to:
       > error: at Jamroot:101
       > error: *** Unable to find Lua library.
       >
       For full logs, run 'nix log /nix/store/phyfv3vv06g2qjm46f8ipvql840bcjva-luabind-0.9.1.drv'.
error: 1 dependencies of derivation '/nix/store/5r0j9xn0w1h3l6bc3vvjabm7bv4828r8-osrm-backend-5.26.0.drv' failed to build
```

### Additional context

FWIW: my fix is here, https://github.com/juspay/nixpkgs/pull/1/files - but it is using a fork of `luabind`.  If somebody has the time to create a minimal patch on top of the current source, please feel free to. 
### Notify maintainers

@erictapen 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.2.0, macOS 13.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by aravindgopall**
###### Description of changes

Fixes #224535 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aravindgopall** [comment]
Fixes #224535

**Comment by erictapen** [comment]
@GrahamcOfBorg build soi icesl

File: pkgs/development/libraries/luabind/default.nix [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+    owner = "Oberon00";
     repo = "luabind";
-    rev = "v${version}";
-    sha256 = "sha256-sK1ca2Oj9yXdmxyXeDO3k8YZ1g+HxIXLhvdTWdPDdag=";
+    rev = "49814f6b47ed99e273edc5198a6ebd7fa19e813a";
+    sha256 = "sha256-JcOsoQHRvdzF2rsZBW6egOwIy7+7C4wy0LiYmbV590Q";
   };
 
-  patches = [ ./0.9.1_modern_boost_fix.patch ./0.9.1_boost_1.57_fix.patch ./0.9.1_discover_luajit.patch ];
+  # patches = [ ./0.9.1_modern_boost_fix.patch ./0.9.1_boost_1.57_fix.patch ./0.9.1_discover_luajit.patch ./0.9.1_darwin_dylib.patch ];
```
Comment (srid) [inline pkgs/development/libraries/luabind/default.nix:14]: The unused patches can be deleted?

And any reason to keep the commented out code (`# patches = ...`) here?

File: pkgs/development/libraries/luabind/default.nix [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-  buildInputs = [ boost-build lua boost ];
+  buildInputs = [ boost-build lua boost cmake ];
 
   propagatedBuildInputs = [ lua ];
 
-  buildPhase = "LUA_PATH=${lua} bjam release";
+  buildPhase = "LUA_PATH=${lua} cmake .";
 
-  installPhase = "LUA_PATH=${lua} bjam --prefix=$out release install";
+#  installPhase = "LUA_PATH=${lua} bjam --prefix=$out release install";
```
Comment (srid) [inline pkgs/development/libraries/luabind/default.nix:22]: Any reason to keep the commented out `installPhase` here?

File: pkgs/servers/osrm-backend/default.nix [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{lib, stdenv, fetchFromGitHub, cmake, pkg-config, bzip2, libxml2, libzip, boost, lua, luabind, tbb, expat}:
+{lib, stdenv, fetchFromGitHub, cmake, pkg-config, bzip2, libxml2, libzip, boost, lua, luabind,  tbb, expat}:
```
Comment (srid) [inline pkgs/servers/osrm-backend/default.nix:1]: Unnecessary space here.

**Review by srid — COMMENTED** [review]
We may want to squash these into one or two commits per [guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes).

File: pkgs/development/libraries/luabind/default.nix [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
 
   propagatedBuildInputs = [ lua ];
 
-  buildPhase = "LUA_PATH=${lua} bjam release";
+  configurePhase = "cmake -DCMAKE_INSTALL_PREFIX=$out/usr/ .";
+
+  buildPhase = "make";
 
-  installPhase = "LUA_PATH=${lua} bjam --prefix=$out release install";
+  installPhase = "make install";
```
[the remark this answers is not in the document]
Comment (aravindgopall) [reply pkgs/development/libraries/luabind/default.nix:22]: yeah. seems only `buildPhase = cmake .` is enough. Updating the PR.

File: pkgs/development/libraries/luabind/default.nix [hunk 5]
```diff
@@ -1,25 +1,23 @@
...
+  nativeBuildInputs = [ cmake ];
 
   buildInputs = [ boost-build lua boost ];
 
   propagatedBuildInputs = [ lua ];
 
-  buildPhase = "LUA_PATH=${lua} bjam release";
-
-  installPhase = "LUA_PATH=${lua} bjam --prefix=$out release install";
+  buildPhase = "cmake .";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/luabind/default.nix:20]: I think the cmake setup-hook should also do this.

**Comment by srid** [comment]
@erictapen @SuperSandro2000 Looks like all the checks passed. Is there anything else that we should do before merging this PR to `master`?

**Comment by erictapen** [comment]
Yes, as you already mentioned the commit messages should be up to the [guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes). Also one commit for the whole thing would be preferable.

**Review by erictapen — APPROVED** [review]
Thanks!

**Pull request merged**